translate:  tutorial/part-three/index.md (#27)
gatsbyjs/gatsby-pt-BR: docs/tutorial/part-three/index.md

disableTableOfContents: true
---

> Bem-vindo a parte três!

Reviewer: Substituir "Bem-vindo" por "Boas-vindas" para ficar neutro.

## O que veremos nesse tutorial?

Nessa parte, você irá aprender sobre plugins do Gatsby e como criar componentes de layout.

Plugins do Gatsby são pacotes JavaScript que ajudam a adicionar funcionalidades a um site Gatsby. O Gatsby foi projetado para ser extensível, o que significa que os plugins podem estender e modificar praticamente qualquer coisa que o Gatsby faz.

> Componentes de layout são para seções do seu site que você quer compartilhar entre diversas páginas. Por exemplo, normalmente sites tem componentes de layout para um cabeçalho e rodapé que são compartilhados. Outras coisas comuns de se adicionar em um layout são *sidebars* e/ou menu de navegação. Nesta página, por exemplo, o cabeçalho no topo é parte de um componente de layout do gatsbyjs.org.

Reviewer 2: Para manter o padrão do restante das documentações, sugiro que deixe em itálico as palavras não traduzidas do inglês.

```suggestion
Componentes de layout são para seções do seu site que você quer compartilhar entre diversas páginas. Por exemplo, normalmente sites tem componentes de layout para um cabeçalho e rodapé que são compartilhados. Outras coisas comuns de se adicionar em um layout são _sidebars_ e/ou menu de navegação. Nesta página, por exemplo, o cabeçalho no topo é parte de um componente de layout do gatsbyjs.org.
```

Vamos nos aprofundar na parte três.

## Usando plugins

Você provavelmente tem familiaridade com o conceito de plugins. Diversos sistemas de software suportam adição de plugins customizados para acrescentar novas funcionalidades ou, até mesmo, modificar o funcionamento interno do software. Os plugins do Gatsby funcionam da mesma maneira.

Membros da comunidade (como você!) podem contribuir com plugins (pequenas quantidades de código JavaScript) que outros podem utilizar quando estiverem construindo sites com Gatsby.

> Existem centenas de plugins! Explore a [Biblioteca de Plugins](/plugins) do Gatsby.

> Nosso objetivo com plugins é que eles sejam simples de instalar e utilizar. Provavelmente você irá utilizar plugins em qualquer site Gatsby que você for construir. Enquanto você estiver percorrendo o restante do tutorial você terá muitas oportunidades para praticar a instalação e utilização de plugins.

Reviewer 3: ```suggestion
Nosso objetivo com plugins é que eles sejam simples de instalar e utilizar. Provavelmente você irá utilizar plugins em quase todos os sites Gatsby que você for construir. Enquanto você estiver percorrendo o restante do tutorial você terá muitas oportunidades para praticar a instalação e utilização de plugins.
```

Para uma introdução inicial em como utilizar plugins, vamos instalar e implementar o plugin Gatsby para Typography.js

[Typography.js](https://kyleamathews.github.io/typography.js/) é uma biblioteca JavaScript que gera estilos base de tipografia para o seu site. A biblioteca tem um [plugin Gatsby correspondente](/packages/gatsby-plugin-typography/) para simplificar o seu uso em um site Gatsby.

### ✋ Criar um novo site Gatsby

> Como mencionamos na [parte dois](/tutorial/part-two/), a partir desse ponto é, provavelmente, uma boa ideia fecha as janelas do terminal e os arquivos das partes anteriores do tutorial, de modo que as coisas se mantenham organizadas na sua área de trabalho. Em seguida, abra uma nova janela do terminal e rode os seguintes comandos para criar um novo site Gatsby em um novo diretório chamado `tutorial-part-three` e entrar nesse diretório.

Reviewer 2: "fecha" => "fechar"
"rode" => "execute"

veja nossas traduções padrões em nosso [Google Spreadsheets](https://docs.google.com/spreadsheets/d/1TLztI-jrPslCt0fgk4B9tGnhhS_7JQZJfMql1ePsIZo/edit#gid=0)

Reviewer 4: de modo que as coisas se mantenham -> que as coisas mantenham-se

}
```

> O `gatsby-config.js` é outro arquivo especial que o Gatsby irá reconhecer automaticamente. É nesse arquivo que você adiciona plugins e outras configrações do site.

Reviewer 5: ```suggestion
O `gatsby-config.js` é outro arquivo especial que o Gatsby irá reconhecer automaticamente. É nesse arquivo que você adiciona plugins e outras configurações do site.
```

> O `gatsby-config.js` é outro arquivo especial que o Gatsby irá reconhecer automaticamente. É nesse arquivo que você adiciona plugins e outras configrações do site.

Reviewer 5: ```suggestion
O `gatsby-config.js` é outro arquivo especial que o Gatsby irá reconhecer automaticamente. É nesse arquivo que você adiciona plugins e outras configurações do site.
```

> Se deseja saber mais, confira a [documentação sobre gatsby-config.js](/docs/gatsby-config/).

> 2. Typography.js precisa de um arquivo de configuração. Cria uma nova pasta chamada `utils` dentro da pasta `src`. Em seguida, adicione um novo arquivo chamado `typography.js` na pasta `utils` e copie o seguinte conteúdo para dentro do arquivo:

Reviewer 3: ```suggestion
3. Typography.js precisa de um arquivo de configuração. Crie uma nova pasta chamada `utils` dentro da pasta `src`. Em seguida, adicione um novo arquivo chamado `typography.js` na pasta `utils` e copie o seguinte conteúdo para dentro do arquivo:
```

)
```

> Agora seu site deve se paracer com isso:

Reviewer 5: ```suggestion
Agora seu site deve se parecer com isso:
```

![no-layout](no-layout.png)

> Vamos fazer uma pequena melhoria. Muitos sites tem uma coluna unica de texto centralizadas no centro da página. Para criar isso adicione os seguintes estilos na

Reviewer 2: "unica" => "única"

![with-layout2](with-layout2.png)

Maravilha. Você instalou e configurou seu primeiríssimo plugin do Gatsby!

## Criando componentes de layout

> Agora vamos aprender sobre componentes de layout. Para se preparar para essa parte, adicione algumas novas páginas no seu projeto: uma página *sobre* e uma página de *contato*

Reviewer 3: ```suggestion
Agora vamos aprender sobre componentes de layout. Para se preparar para essa parte, adicione algumas novas páginas no seu projeto: uma página _sobre_ e uma página de _contato_
```

)
```

> Vamos ver como a nova página *sobre* ficou:

Reviewer 3: ```suggestion
Vamos ver como a nova página _sobre_ ficou:
```
Creio que seja melhor usar o italico para evidenciar, no lugar do negrito

Author: Na verdade, os dois casos estão em itálico. Negríto é `**texto**`, itálico pode ser `_texto_` ou `*texto*`. Mas vou aceitar a sugestão pra manter no padrão do projeto.

Reviewer 3: verdade, erro meu

![about-uncentered](about-uncentered.png)

> Hmm. Seria legal se o conteúdo das duas páginas fossem centralizados como na página inicial. E também serial legal se tivéssemos alguma navegação global para que os visitantes encontrem e visitem as outras páginas.

Reviewer 3: ```suggestion
Hmm. Seria legal se o conteúdo das duas páginas fossem centralizados como na página inicial. E também serial legal se tivéssemos alguma navegação global, facilitando que os novos visitantes encontrem e visitem as outras páginas.
```
O que tu acha dessa sugestão, eu sugeri para facilitar a leitura, o que tu acha?

![with-layout2](with-layout2.png)

> Que ótimo, o layout está funcionado! O conteúdo da sua página incial ainda está centralizado.

Reviewer 3: ```suggestion
Que ótimo, o layout está funcionando! O conteúdo da sua página inicial ainda está centralizado.
```

)
```

> Se você for a qualquer uma das suas três páginas, você verá o mesmo título adicionado, por exemplo como na página `/about/`:

Reviewer 3: ```suggestion
Se você for a qualquer uma das suas três páginas, você verá o mesmo título adicionado, como, por exemplo, na página `/about/`:
```

```

If you go to any of your three pages, you'll see the same title added, e.g. the `/about/` page:

> Se você for a qualquer uma das suas três páginas, você verá o mesmo título adicionado, por exemplo como na página `/about/`:

Reviewer 3: Falta remover o texto em inglês

![with-navigation2](with-navigation2.png)

E aí está! Um site com três páginas e navegação global básica.

_Desafio:_ Com os poderes do seu novo "componente de layout", tente adicionar cabeçalho, rodapé, navegação global, sidebar, etc. nos seus sites Gatsby.

## O que vem a seguir?

> Continue na [parte quatro do tutorial](/tutorial/part-four/) onde você irá começar a aprenter sobre a camada de dados do Gatsby e como criar páginas programaticamente.

Reviewer 2: "aprenter" => "aprender"

> Continue na [parte quatro do tutorial](/tutorial/part-four/) onde você irá começar a aprender sobre a camada de dados do Gatsby e como criar páginas programaticamente.

Reviewer 3: ```suggestion
Continue para a [parte quatro do tutorial](/tutorial/part-four/) onde você irá começar a aprender sobre a camada de dados do Gatsby e como criar páginas programaticamente.
```
tu acha que fica melhor?

Author: Nesse caso eu acho que `na` faz mais sentido 🤔 
acho que `para` leria melhor se fosse algo como `siga para a parte 4...`

Reviewer 3: Pode ser esse debaixo então hehe
